src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksReportServiceImpl.java

```diff
@@ -0,0 +1,270 @@
+package edu.cornell.kfs.pmw.batch.service.impl;
+
+import java.lang.String;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.kuali.rice.core.api.config.property.ConfigurationService;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.sys.KFSConstants;
+
+import edu.cornell.kfs.pmw.batch.PaymentWorksKeyConstants;
+import edu.cornell.kfs.pmw.batch.PaymentWorksParameterConstants;
+import edu.cornell.kfs.pmw.batch.businessobject.PaymentWorksVendor;
+import edu.cornell.kfs.pmw.batch.report.PaymentWorksBatchReportVendorItem;
+import edu.cornell.kfs.pmw.batch.service.PaymentWorksBatchUtilityService;
+import edu.cornell.kfs.pmw.batch.service.PaymentWorksReportEmailService;
+import edu.cornell.kfs.pmw.batch.service.PaymentWorksReportService;
+import edu.cornell.kfs.sys.service.ReportWriterService;
+
+public abstract class PaymentWorksReportServiceImpl implements PaymentWorksReportService {
+    private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(PaymentWorksReportServiceImpl.class);
+    
+    protected ConfigurationService configurationService;
+    protected PaymentWorksBatchUtilityService paymentWorksBatchUtilityService;
+    protected PaymentWorksReportEmailService paymentWorksReportEmailService;
+    protected ReportWriterService reportWriterService;
+    
+    protected String toAddress = null;
+    protected String fromAddress = null;
+    protected String reportFileNamePrefix = null;
+    protected String reportTitle = null;
+    protected String summarySubTitle = null;
+    protected String processedSubTitle = null;
+    protected String processingErrorsSubTitle = null;
+    protected String unprocessedSubTitle = null;
+    protected String paymentWorksVendorIdLabel = null;
+    protected String submittedDateLabel = null;
+    protected String vendorTypeLabel = null;
+    protected String vendorNameLabel = null;
+    protected String taxIdTypeLabel = null;
+    protected String vendorSubmitterEmailLabel = null;
+    protected String initiatorNetidLabel = null;
+    protected String errorsLabel = null;
+    
+    public void sendEmailThatNoDataWasFoundToProcess(List<String> emailSubjectItems, List<String> emailBodyItems) {
+        LOG.info("sendEmailThatNoDataWasFoundToProcess: Preparing to send email that no data was found to process.");
+        String body = constructStringFromListItemsDoubleSepartedByDelimitter(emailBodyItems, System.lineSeparator());
+        String subject = constructStringFromListItemsDoubleSepartedByDelimitter(emailSubjectItems, KFSConstants.BLANK_SPACE);
+        getPaymentWorksReportEmailService().sendEmail(getToAddress(), getFromAddress(), subject, body);
+        LOG.info("sendEmailThatNoDataWasFoundToProcess: Email was sent that no data was found to process.");
+    }
+    
+    private String constructStringFromListItemsDoubleSepartedByDelimitter(List<String> itemsToUse, String delimitter) {
```

Comment:
Another method with minor typos in the name; given that this is a private method only used by this class, deferring a fix to a future story is okay with me.